akka-projection-cassandra/src/test/scala/akka/projection/cassandra/CassandraProjectionSpec.scala

```diff
@@ -806,6 +810,34 @@ class CassandraProjectionSpec
       handlerProbe.expectMessage(handler.failedMessage)
       handlerProbe.expectNoMessage() // no duplicate stop
     }
+
+    "be able to stop when retrying" in {
+      val entityId = UUID.randomUUID().toString
+      val projectionId = genRandomProjectionId()
+
+      val handlerProbe = createTestProbe[String]()
+      val handler = new LifecycleHandler(handlerProbe.ref, alwaysFailOnOffset = 4)
+
+      val projection =
+        CassandraProjection
+          .atLeastOnce[Long, Envelope](projectionId, sourceProvider(system, entityId), handler)
+          .withRecoveryStrategy(HandlerRecoveryStrategy.retryAndFail(100, 100.millis))
+          .withSaveOffset(1, Duration.Zero)
+
+      val ref = spawn(ProjectionBehavior(projection))
+
+      handlerProbe.expectMessage(handler.startMessage)
+      handlerProbe.expectMessage("abc")
+      handlerProbe.expectMessage("def")
+      handlerProbe.expectMessage("ghi")
+      // fail 4
+
+      // let it retry for a while
+      Thread.sleep(300)
```

Comment:
Maybe this could be an `handlerProbe.expectNoMessage` for some timeout?

Reply:
The Lifecycle handler does send a 'failed' message on each failure. So we will see a message passing. 

We could use it to prove that indeed is failing (at least once) and because `retryAndFail(100, 100.millis)` we know it keeps trying.

Reply:
there was something here that didn't work with, but I can try again